### PR TITLE
feat: Add RPC client/server for EVM methods

### PR DESCRIPTION
**Pull request by amaury1093**
Related to https://github.com/Inco-fhevm/inco-monorepo/pull/20

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/keeper/state_transition.go
Comment (KentaMori115) [inline x/evm/keeper/state_transition.go]: @amaury1093 I guess there may need to return an handler or identity in order to specify which evm instance we should use in the latter rcp calls.

File: x/evm/keeper/state_transition.go
Comment (KentaMori115) [inline x/evm/keeper/state_transition.go]: @amaury1093 I think the param `args` should contain evm (sgx side) handle ID that will be interacted.

File: x/evm/keeper/state_transition.go
Comment (amaury1093) [reply x/evm/keeper/state_transition.go]: Why do we need to specify which EVM instance? Txs will be run sequentially, so there will only be 1 instance at a time

**Review by KentaMori115 — APPROVED** [review]
LGTM.

File: x/evm/keeper/state_transition.go
Comment (amaury1093) [reply x/evm/keeper/state_transition.go]: See https://github.com/Inco-fhevm/ethermint/pull/2#discussion_r1543080522.

However, thinking more about it, it might be a good idea to still have some kind of ID per EVM instance. The ID can be the TxHash also.

**Comment by amaury1093** [comment]
Before merging this PR, I'd like to test it with a RPC server on the SGX side too. @kenta-mori3322 LMK about your progress there.

**Review by lovenoble — APPROVED** [review]

File: x/evm/keeper/state_transition.go
Comment (amaury1093) [inline x/evm/keeper/state_transition.go:563]: Can you explain why we need a custom encoding here?

File: x/evm/keeper/state_transition.go
Comment (amaury1093) [inline x/evm/keeper/state_transition.go:570]: I know I created this function, but I think it's super ugly to put here.

IMO, we should run the RPC server in `server/start.go`, like all other servers. Then register this RPC server in `RegisterServices` or something like that.

File: x/evm/keeper/state_transition.go
Comment (KentaMori115) [reply x/evm/keeper/state_transition.go:563]: @amaury1093 cfg.Overried isn't serializable.

File: x/evm/keeper/state_transition.go
Comment (KentaMori115) [reply x/evm/keeper/state_transition.go:570]: @amaury1093 let me try.

File: x/evm/keeper/state_transition.go
Comment (amaury1093) [reply x/evm/keeper/state_transition.go:563]: I'm not 100% ToGob64 works. This test fails. Could you double check on your side too @kenta-mori3322?

```go
func TestABC(t *testing.T) {
	a := types.StateOverride{}
	myNum := hexutil.Big(*big.NewInt(20))
	pMyNum := &myNum
	addr := common.HexToAddress("0x9F907cD9429939D2b41CED2950C996b9D8481B32")
	a[addr] = types.OverrideAccount{
		Balance: &pMyNum,
	}
	str := keeper.ToGOB64(&a)
	fmt.Println(str)
	b := keeper.FromGOB64(str)
	require.Equal(t, a[addr].Balance, (*b)[addr].Balance)
}

```

I had the same issue with `cfg.ChainConfig`, and taking a look at geth code they support well JSON serialization. It seems that this OverrideAccount is also json-serialiazable, maybe we can switch to that?

**Comment by kenta-mori3322** [comment]
Gotcha. let me take a look at it.

File: x/evm/keeper/state_transition.go
Comment (KentaMori115) [reply x/evm/keeper/state_transition.go:563]: @amaury1093 Gotcha, let me take a look at it.

**Comment by amaury1093** [comment]
Merging to unblock work on multi-keys

**Pull request merged**
